src/composer/src/passes.rs

```diff
@@ -129,7 +129,12 @@ impl SystemState {
 // that file already exists. Use unique names for files, and use the
 // component-namespacing of names for per-component files.
 pub trait BuildState {
-    fn initialize(&mut self, name: &String, s: &SystemState) -> Result<(), String>; // must be called *before* the following functions
+    fn initialize(
+        &mut self,
+        name: &String,
+        flag: Option<String>,
```

Comment:
`flag` doesn't really give any information as to what the argument is. Better argument name?

Reply:
we change the name from flag to is_rebuild